src/absil/ilreflect.fs

```diff
@@ -1235,9 +1271,12 @@ let emitCode cenv modB emEnv (ilG:ILGenerator) code =
 let emitLocal cenv emEnv (ilG : ILGenerator) (local: ILLocal) =
     let ty = convType cenv emEnv  local.Type
     let locBuilder = ilG.DeclareLocalAndLog(ty, local.IsPinned)
+#if NO_PDB_WRITER
```

Comment:
Rather than having another `#if` why not just throw if a PDB is requested on CoreCLR?  This would reduce the complexity of the code (it's the approach the other managed compilers are taking).